source/Core/Threads/GUIThread.cpp

```diff
@@ -860,13 +860,12 @@ void startGUITask(void const *argument) {
   getTipRawTemp(1); // reset filter
   OLED::setRotation(getSettingValue(SettingsOptions::OrientationMode) & 1);
   uint32_t ticks = xTaskGetTickCount();
-  ticks += (TICKS_SECOND * 4); // 4 seconds from now
-  while (xTaskGetTickCount() < ticks) {
+  ticks += (getSettingValue(SettingsOptions::LOGOTime) / 100 * TICKS_SECOND);
+  while ((getSettingValue(SettingsOptions::LOGOTime) > 0 && xTaskGetTickCount() < ticks)) {
```

Comment:
Rather than evaluating this in the loop, just throw an if statement around it :)